### PR TITLE
Add typing coverage to mysql providers package

### PR DESCRIPTION
**Pull request by rafyzg**
related: #9708 
Adding typing annotations to mysql providers package.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/providers/mysql/hooks/mysql.py [hunk 1]
```diff
@@ -182,7 +184,7 @@ def bulk_dump(self, table, tmp_file):
         conn.commit()
 
     @staticmethod
-    def _serialize_cell(cell, conn):  # pylint: disable=signature-differs
+    def _serialize_cell(cell: object, conn: Connection) -> object:  # pylint: disable=signature-differs
```
Comment (potiuk) [inline airflow/providers/mysql/hooks/mysql.py]: Maybe we can also match the signatures. I believe the "signature-differs" pylint exception can be removed by adding = None for conn parameter. Which is no harm at all.

**Comment by turbaszek** [comment]
@rafyzg can you please take a look at CI static checks errors?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Just one static check to fix

**Comment by potiuk** [comment]
Stil a static code check :(. I really recomend using pre-commits. They save a ton of iteration overhead :) (see STATIC_CHECKS.rst)

**Comment by potiuk** [comment]
Thanks @rafyzg ! What's next :) ?

**Pull request merged**